src/elements/fields/PaymentMethodField.jsx

```diff
@@ -0,0 +1,193 @@
+  element,
+  runtime,
+  applyStyles,
+  fieldLabel,
+  elementProps = {},
+  setCardElement = () => {},
+  setFieldError = () => {},
+  onChange = () => {},
+  editable = false,
+  inlineError,
+  errorDisplayMode
+}) => {
+  const stripe = useStripe();
+  const elements = useElements();
+
+  useEffect(() => {
+    if (!stripe || !elements) {
+      // Stripe.js has not yet loaded.
+      return;
+    }
+    setCardElement(element.servar.key, elements.getElement(CardElement));
+  }, [stripe, element.servar.key, elements]);
+
+  const cardElementOptions = {
+    style: {
+      base: {
+        textDecoration: 'none', // Bug in card element - force the reset to none
+        ...filterKeys(applyStyles.getTarget('field'), supportedStripeCssProps),
+        '::placeholder': {
+          textDecoration: 'none', // Bug in card element - force the reset to none
+          ...filterKeys(
+            applyStyles.getTarget('placeholder'),
+            supportedStripeCssProps
+          )
+        },
+        ':hover': filterStyleValues(
+          filterKeys(applyStyles.getTarget('hover'), supportedStripeCssProps),
+          '!important'
+        ),
+        ':focus': filterStyleValues(
+          filterKeys(applyStyles.getTarget('active'), supportedStripeCssProps),
+          '!important'
+        )
+      },
+      complete: {
+        ...applyStyles.getTarget('completed')
+      },
+      invalid: {
+        color: '#d6504a',
+        iconColor: '#d6504a'
+      }
+    },
+    classes: {
+      base: element.servar.key
+    }
+  };
+  const handleCardChange = async (event) => {
+    // Listen for changes in the CardElement
+    // and display any errors as the customer types their card details
+    setFieldError(event.error?.message ?? '');
+    // also set the field data with complete=true so we know
+    // when a card number has been entered and is complete
+    onChange(
+      event.complete
+        ? { complete: true, card_data: { brand: event.brand } }
+        : ''
+    );
+  };
+
+  return (
+    <div
+      css={{
+        maxWidth: '100%',
+        position: 'relative',
+        pointerEvents: editable ? 'none' : 'auto',
+        ...applyStyles.getTarget('fc')
+      }}
+      {...elementProps}
+    >
+      {fieldLabel}
+      <div
+        css={{
+          position: 'relative',
+          width: '100%',
+          whiteSpace: 'nowrap',
+          overflowX: 'hidden',
+          padding: '0 2px',
+          display: 'flex',
+          alignItems: 'center',
+          ...applyStyles.getTarget('sub-fc'),
+          '&:focus': applyStyles.getTarget('active'),
+          '&:hover': applyStyles.getTarget('hover'),
+          ...(inlineError ? { borderColor: '#F42525' } : {})
+        }}
+      >
+        <div css={{ width: '100%', position: 'relative' }}>
+          {/* position an input field under the card element to support html5 errors */}
+          {runtime && errorDisplayMode === 'html5' && (
+            <input
```

Comment:
we do something similar with a hdiden input field for hte file _upload element. we might want to unify those components

Reply:
@bo-dun-1 I looked at the input in FileUploadField.  I don't see a lot of commonality other than they both serve as 'anchors' for an html5 error.  The one in FileUploadField actually serves a functional purpose beyond that.  The one in PaymentMethod only serves as the anchor.  So, I just don't see the value in commonizing them.  If you disagree, maybe you can offer some specifics?

Reply:
maybe at least their styles? currently file uses opacity and this one uses background